src/expr/node_manager.cpp

```diff
@@ -961,6 +961,17 @@ Node NodeManager::mkSingleton(const TypeNode& t, const TNode n)
   return singleton;
 }
 
+Node NodeManager::mkBag(const TypeNode& t, const TNode n, const TNode m)
```

Comment:
If we expose SINGLETON, and MK_BAG in the Solver class, I don't think we need mkBag in or mkSingleton. We can combine both in Solver::mkOp<T>(Sort s), and then use the normal function Solver::mkTerm(Op op, Term child)

Reply:
We need this because the internal code should never call the API.